### ISSUE TITLE
Errors from CI

### ISSUE THREAD

**Issue by nmvdw** [UniMath/UniMath#1594]
Yesterday, the CI got the following error:
https://github.com/UniMath/UniMath/actions/runs/3404826185/jobs/5662342283
```
Error: Ill-formed template inductive declaration: not polymorphic on any [2934](https://github.com/UniMath/UniMath/actions/runs/3404826185/jobs/5662342283#step:4:2935)
universe.
```
I get this error on my laptop as well, and the solution was to add `: UU` to some inductive declarations: https://github.com/UniMath/SetHITs/blob/master/code/examples/join_semilattice.v#L26.

So, I did that, and this one managed to pass. Today there's another one:
https://github.com/UniMath/UniMath/actions/runs/3409009563/jobs/5670255679, and now the error is
```File "./SetHITs/code/examples/groups.v", line 118, characters 12-20:
[368](https://github.com/UniMath/UniMath/actions/runs/3404826185/jobs/5662342283#step:7:369)
Error: [369](https://github.com/UniMath/UniMath/actions/runs/3404826185/jobs/5662342283#step:7:370)
The term "group_ax" has type "UU" while it is expected to have type [370](https://github.com/UniMath/UniMath/actions/runs/3404826185/jobs/5662342283#step:7:371)
"Type" (universe inconsistency)."
```
It wants the `: UU` to be removed. So, the solution of the error yesterday is the cause of the error today.

Why does this happen and how should one prevent this?

Note: in GrpdHITs, I can just do the following and it never complains https://github.com/UniMath/GrpdHITs/blob/de5adde2563e1cb90d24b268d2f848106638807c/code/examples/torus.v#L34

**Comment by m-lindgren** [comment]
Interesting. What version of coq are you using? The CI SetHITs is currently using the bundled version which is 8.12-something I think. Should I try to see if a newer version of Coq works better in the CI? Like 8.16.0 or 8.15.2 or something.

It's probably a good idea to upgrade to a modern version anyway, since it seems (for some reason) the makefile stops because the installed dune version is too old (why it cares about dune I don't know).

**Comment by nmvdw** [comment]
I use the version of Coq that comes with UniMath, so 8.12.2 (compiled with OCaml 4.12).

I will try adding `-type-in-type` to SetHITs, and see whether that works.

**Comment by m-lindgren** [comment]
Do you know which commit introduced this error? I just tried compiling [c175809](https://github.com/UniMath/SetHITs/commit/c175809c457c367a3a1cd52668c314c236361c2b) with Coq 8.12.2 and without `-type-in-type` and had no problem, except warnings about gradth.

**Comment by nmvdw** [comment]
The relevant commit is the latest one. The `gradths` are removed in that one as well.

**Comment by m-lindgren** [comment]
Okay, I've finally managed to reproduce the error messages but I had to enable `-type-in-type`. I don't have them when I don't use `-type-in-type`. If I enable that flag I have errors and build failures like this all the way back to July at least. 

I think removing `-type-in-type` and applying the changes in UniMath/SetHITs#14 would fix all the issues but I'll try to verify it. I'll fix the CI at the same time.
